File: scripts/build.10m.js
#!/usr/bin/env /usr/local/bin/node

'use strict';

const got = require('/usr/local/lib/node_modules/got');

const API_TOKEN = 'PUT_TOKEN_HERE';
const belvEndpoint = `https://mr-belvedere.herokuapp.com/api/v1/jobs?api_token=${API_TOKEN}`;

got(belvEndpoint)
  .then(response => {
    const data = JSON.parse(response.body);

    data.map(item => {
      if (item.status == "success") {
        console.log(`${item.name} is stable :thumbsup:`)
      } else if (item.status == "failure") {
        console.log(`${item.name} is failing :thumbsdown: |color=red`)
      } else if (item.stauts == "building") {
        console.log(`${item.name} is building :construction_worker: |color=orange`)
      }
    })

  })
  .catch(() => {
    console.log('Error parsing response');
  });

File: scripts/build.10m.rb
#!/usr/bin/ruby

require "json"
require "open-uri"

request_uri = "https://mr-belvedere.herokuapp.com/api/v1/jobs?api_token=EMAIL ME FOR API KEY"

buffer = open(request_uri).read
results = JSON.parse(buffer, symbolize_names: true)

def parse_result(result)
  case result
  when "success"
    return "stable :ok_hand: | color=green"
  when "failure"
      return "failing :sob: | color=red"
  when "building"
    return "building :construction_worker: | color=orange"
  end
end

results.each do |result|
  status = parse_result result[:status]
  puts "#{result[:name]} is #{status} "
end


